store/governance/proposal.js

```diff
@@ -0,0 +1,257 @@
+const CACHE_BLOCK = {}
+
+const parseComment = (calldata, govInstance) => {
+  const empty = { contact: '', message: '' }
+  if (!calldata || !govInstance) return empty
+
+  const methodLength = 4 // length of castDelegatedVote method
+  const result = utils.defaultAbiCoder.decode(
+    ['address[]', 'uint256', 'bool'],
+    utils.hexDataSlice(calldata, methodLength)
+  )
+  const data = govInstance.methods.castDelegatedVote(...result).encodeABI()
+  const dataLength = utils.hexDataLength(data)
+
+  try {
+    const str = utils.defaultAbiCoder.decode(['string'], utils.hexDataSlice(calldata, dataLength))
+    const [contact, message] = JSON.parse(str)
+    return { contact, message }
+  } catch {
+    return empty
+  }
+}
+
+const createProposalComment = (resultAll, votedEvent) => {
+  const { transactionHash, returnValues, blockNumber } = votedEvent
+  const { voter } = returnValues
+
+  const comment = parseComment()
+
+  const percentage =
+    toBN(votedEvent.returnValues.votes)
+      .mul(toBN(10000))
+      .divRound(resultAll)
+      .toNumber() / 100
+
+  return {
+    id: `${transactionHash}-${voter}`,
+    percentage,
+    ...returnValues,
+    votes: fromWei(returnValues.votes),
+    transactionHash,
+    blockNumber,
+
+    ...comment,
+
+    ens: {
+      delegator: null,
+      voter: null
+    },
+    delegator: null,
+    timestamp: null
+  }
+}
+
+const createFetchCommentsWithMessages = (web3, batch, govInstance) => async (proposalComment) => {
+  const { transactionHash, voter, blockNumber } = proposalComment
+
+  if (!CACHE_TX[transactionHash]) {
+    CACHE_TX[transactionHash] = new Promise((resolve, reject) => {
+      const callback = createBatchRequestCallback(resolve, reject)
+      batch.add(web3.eth.getTransaction.request(transactionHash, callback))
+    })
+  }
+
+  if (!CACHE_BLOCK[blockNumber]) {
+    CACHE_BLOCK[blockNumber] = new Promise((resolve, reject) => {
+      const callback = createBatchRequestCallback(resolve, reject)
+      batch.add(web3.eth.getBlock.request(blockNumber, callback))
+    })
+  }
+
+  try {
+    const [tx, blockInfo] = await Promise.all([CACHE_TX[transactionHash], CACHE_BLOCK[blockNumber]])
+
+    const isMaybeHasComment = voter === tx.from
+    const comment = parseComment(isMaybeHasComment ? tx.input : null, govInstance)
+
+    return {
+      ...proposalComment,
+      ...comment,
+
+      delegator: voter === tx.from ? null : tx.from,
+      timestamp: blockInfo.timestamp
+    }
+  } catch (error) {
+    CACHE_TX[transactionHash] = null
+    CACHE_BLOCK[blockNumber] = null
+    return proposalComment
+  }
+}
+
+const state = () => {
+  return {
+    isFetchingComments: false,
+    isFetchingMessages: false,
+    ensNames: {},
+    comments: []
+  }
+}
+
+const getters = {
+  comments: (state) => {
+    const { ensNames } = state
+    let comments = state.comments.slice()
+
+    comments.sort((a, b) => b.blockNumber - a.blockNumber)
+    comments = uniqBy(comments, 'voter')
```

Comment:
use uniqBy before sorting

Reply:
it's not the same thing, the order is important